src/jemalloc.c

```diff
@@ -3124,16 +3144,7 @@ je_mallocx(size_t size, int flags) {
 
 		dopts.zero = MALLOCX_ZERO_GET(flags);
 
-		if ((flags & MALLOCX_TCACHE_MASK) != 0) {
-			if ((flags & MALLOCX_TCACHE_MASK)
-			    == MALLOCX_TCACHE_NONE) {
-				dopts.tcache_ind = TCACHE_IND_NONE;
-			} else {
-				dopts.tcache_ind = MALLOCX_TCACHE_GET(flags);
-			}
-		} else {
-			dopts.tcache_ind = TCACHE_IND_AUTOMATIC;
-		}
+		dopts.tcache_ind = mallocx_tcache_get(flags);
```

Comment:
In this case, the branch prediction inside `mallocx_tcache_get()` becomes redundant to some extent, given the `unlikely(flags != 0)` guard above, but it likely would not do harm.